main/lsp/notifications/did_change_configuration.cc

```diff
@@ -0,0 +1,30 @@
+#include "main/lsp/notifications/did_change_configuration.h"
+#include "main/lsp/LSPIndexer.h"
+#include "main/lsp/LSPPreprocessor.h"
+
+namespace sorbet::realmain::lsp {
+DidChangeConfigurationTask::DidChangeConfigurationTask(const LSPConfiguration &config,
+                                                       std::unique_ptr<DidChangeConfigurationParams> params)
+    : LSPTask(config, LSPMethod::WorkspaceDidChangeConfiguration), params(move(params)), openFilePaths(), epoch(1) {}
+
+LSPTask::Phase DidChangeConfigurationTask::finalPhase() const {
+    // We want this to run all the way so that the changes to
+    // Global State get propagated through.
+    return LSPTask::Phase::RUN;
+}
+
+void DidChangeConfigurationTask::preprocess(LSPPreprocessor &preprocessor) {
+    auto tmp = preprocessor.openFilePaths();
+    openFilePaths = std::move(tmp);
+    epoch = preprocessor.incNextVersion();
```

Comment:
Why expose this function, versus accepting `epoch` as an argument to the `DidChangeConfigurationTask` constructor, and populating it via `nextVersion++` in the `getTaskForMessage` method like everywhere else?